src/Compilers/VisualBasic/Test/Semantic/Semantics/NameOfTests.vb

```diff
@@ -3590,5 +3590,49 @@ End Module
             Dim comp = CreateCompilationWithMscorlibAndVBRuntime(compilationDef, options:=TestOptions.DebugExe, additionalRefs:={LinqAssemblyRef})
             CompileAndVerify(comp, expectedOutput:="xxx").VerifyDiagnostics()
         End Sub
+
+        <Fact, WorkItem(10839, "https://github.com/dotnet/roslyn/issues/10839")>
+        Public Sub ForbidInstanceQualifiedFromTypeInNestedExpression()
+            Dim compilationDef =
+                <compilation>
+                    <file name="a.vb">
+Class C
+    Public MyInstance As String
+    Public Shared MyStatic As String
+
+    Sub M()
+        Dim X As String
+        X = NameOf(C.MyInstance)
```

Comment:
Can we verify the values like in C#, for those that are legal?

Reply:
This test is only intended to demonstrate which cases are illegal. I believe we already have tests for the string produced by `NameOf`.